### PR TITLE
FEAT: Coding challenge

### PR DESCRIPTION
**Pull request by gasperlf**

### CODE REVIEW

File: src/main/java/com/wezaam/withdrawal/application/impl/WithdrawalServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    @Override
+    @Transactional
+    public Withdrawal create(CreateWithdrawalCommand createWithdrawalCommand) {
+
+        User user =
+                userRepository
+                        .findById(createWithdrawalCommand.getUserId())
+                        .orElseThrow(() -> new NotFoundException("User not found"));
+
+        if (WithdrawalType.ASAP.name().equals(createWithdrawalCommand.getType().name())) {
```
Comment (makcon) [inline src/main/java/com/wezaam/withdrawal/application/impl/WithdrawalServiceImpl.java:49]: Hey @gasperlf, have you consider megring `Withdrawal` and `WithdrawalScheduled`?

Comment (gasperlf) [reply src/main/java/com/wezaam/withdrawal/application/impl/WithdrawalServiceImpl.java:49]: Yes, I did.
but at the end I've prioritize code refactor  to keep functionality and then go with other changes.

Comment (gasperlf) [reply src/main/java/com/wezaam/withdrawal/application/impl/WithdrawalServiceImpl.java:49]: And with the new architecture base on domain it is possible to extend o segregate withdrawal

Comment (makcon) [reply src/main/java/com/wezaam/withdrawal/application/impl/WithdrawalServiceImpl.java:49]: Ok, let's discuss it tomorrow :)

**Pull request closed without merging**